src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -392,6 +392,19 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
 
     if ((enableHttpCache || enableDiskCache) && !enableGrpcCache) {
       initHttpAndDiskCache(env, credentials, authAndTlsOptions, remoteOptions, digestUtil);
+      buildEventArtifactUploaderFactoryDelegate.init(
+        new ByteStreamBuildEventArtifactUploaderFactory(
+          executorService,
+          env.getReporter(),
+          verboseFailures,
+          actionContextProvider.getRemoteCache(),
+          remoteOptions.remoteInstanceName,
+          remoteOptions.remoteBytestreamUriPrefix,
+          buildRequestId,
+          invocationId,
+          remoteOptions.remoteBuildEventUploadMode
+        )
+      );
```

Comment:
As written, this will also cause BEP-referenced files to be uploaded to a disk cache. Is this an intentional part of this PR? (I'm not super convinced that uploading these files to the disk cache is useful, but on the other hand, it's awkward to get a reference to "just the remote cache" here in the combined disk+http cache case, so I could be convinced to leave it in.)